doc/Learn/Mvux/WorkingWithRecords.md

```diff
@@ -0,0 +1,125 @@
+
+# How to write Records with MVUX
+
+## What is a Record
+
+A [record](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record) behaves like a class*, offering the feature of **immutability**, where the values assigned to it remain unchanged once set. It's possible to create records using the `record` modifier, for example:
+
+```csharp
+public record MyRecord();
+```
+
+Records **can** be, but are not necessarily, immutable. Also, records introduce the `with` operator, which is a helpful tool to deal with immutable objects, we will see more about this operator in the **Updating records** section.
+
+> [!IMPORTANT]
+> \* When using `record struct`, there are some differences in how it behaves compared to regular records or classes because it combines value-type characteristics with the features of records. Learn more about [`struct`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct).
+
+## Why Immutability in MVUX
+
+Immutability is crucial in MVUX for two main reasons:
+
+### Predictable State Changes
+
+Immutability ensures that once we set a state, it can't be changed. This predictability makes it easier to understand how our application behaves over time. In MVUX, we use immutable data structures to represent the application state.
+
+### Concurrency and Threading
+
+Immutability makes our application more robust in handling concurrency and threading challenges. Immutable data structures are naturally thread-safe, reducing the risk of bugs related to multiple things happening simultaneously in our app.
+
+## How to create immutable records
+
+You can create immutable records in three ways. First, declare your record with a [primary constructor](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#primary-constructors) and parameters; this will create an immutable record with the specified parameters as its properties:
+
+```csharp
+public partial record ChatResponse(string Message, bool IsError);
+```
+
+The second way is by creating properties using the `init` keyword instead of `set` to enforce immutability. Here's a brief example:
+
+```csharp
+public partial record ChatResponse
+{
+    public string Message { get; init; }
+    public bool IsError { get; init; }
+}
+```
+
+Finally, it's possible to mix the two approaches by using the primary constructor and adding properties for non-essential values. For example, consider a `ChatResponse` record where `Message` is essential, but `IsError` is not:
+
+```csharp
+public partial record ChatResponse(string Message)
+{
+    public bool IsError { get; init; }
+}
+```
+
+```csharp
+new ChatResponse("Hello, I'm a bot"); //with IsError defaulting to false
+```
+
+When you create a record, if you let the properties change with the `set` keyword, the record won't be immutable. This means you won't be able to lock in or keep the values from changing once you've set them.
+
```

Comment:
We might also add a word about the fact that sub properties / objects should also be immutable.

Reply:
Changes in next commit.